x/gov/errors.go

```diff
@@ -0,0 +1,64 @@
+//nolint
+package gov
+
+import (
+	"strconv"
+
+	"encoding/hex"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+const (
+	DefaultCodespace sdk.CodespaceType = 5
+
+	// Gov errors reserve 401 ~ 499.
+	CodeUnknownProposal         sdk.CodeType = 1
+	CodeInactiveProposal        sdk.CodeType = 2
+	CodeAlreadyActiveProposal   sdk.CodeType = 3
+	CodeAlreadyFinishedProposal sdk.CodeType = 4
+	CodeAddressNotStaked        sdk.CodeType = 5
+	CodeInvalidTitle            sdk.CodeType = 6
+	CodeInvalidDescription      sdk.CodeType = 7
+	CodeInvalidProposalType     sdk.CodeType = 8
+	CodeInvalidVote             sdk.CodeType = 9
+)
+
+//----------------------------------------
+// Error constructors
+
+func ErrUnknownProposal(proposalID int64) sdk.Error {
```

Comment:
These error functions need to take a `codespace` as a parameter, not use the default codespace - see `x/stake/errors.go` for an example